test_driver/cloud_firestore_behaviors.dart

```diff
@@ -91,14 +91,114 @@ void main() {
       await doc.setData(<String, dynamic>{
         'message': 'hello firestore',
         'created_at': currentDateTime,
+        'nested1': {
+          'field2': 2,
+          'nested2': {
+            'field3': 3,
+            'nested3': {'field4': 4}
+          }
+        }
       });
 
       final result = await doc.get();
 
       await doc.delete();
 
-      expect(result.data['message'], 'hello firestore');
       expect(result.documentID, documentId);
+      expect(result.data['message'], 'hello firestore');
+      final map1 = result.data['nested1'] as Map<String, dynamic>;
+      expect(map1['field2'], 2);
+      final map2 = map1['nested2'] as Map<String, dynamic>;
+      expect(map2['field3'], 3);
+      final map3 = map2['nested3'] as Map<String, dynamic>;
+      expect(map3['field4'], 4);
+    });
+
+    ftest('Documents should be saved saparately', (firestore) async {
```

Comment:
This is pretty much the same as the one in cloud_firestore_mocks_test.dart.

Reply:
(btw, you left a typo)

Reply:
Fixed.